### PR TITLE
[ROCm]: Updating Matrix Solve Op to use rocSolver

### PR DESCRIPTION
**Pull request by stevenireeves**
Changes in this PR:

1. Adding rocSolver functions to be used in matrix solve op
              CudaSolver and ROCmSolver classes -> GpuSolver class
2.  Merged cuda_solvers.h and rocm_solvers.h into gpu_solvers.h

### CODE REVIEW

**Comment by stevenireeves** [comment]
@penpornk Can you take a look?

**Comment by deven-amd** [comment]
@chsigg @cheshire gentle ping

**Comment by stevenireeves** [comment]
@chsigg @cheshire @penpornk ping.

**Comment by stevenireeves** [comment]
@chsigg @cheshire @penpornk ping

**Review by cheshire — APPROVED** [review]

**Comment by stevenireeves** [comment]
@gbaned Any updates on this PR?

File: tensorflow/core/kernels/linalg/matrix_solve_op.cc
Comment (cheshire) [inline tensorflow/core/kernels/linalg/matrix_solve_op.cc:309]: what's the difference between CUDA/ROCm here? Just `const`? I would drop it for both then.

Comment (stevenireeves) [reply tensorflow/core/kernels/linalg/matrix_solve_op.cc:309]: Resolved.

**Comment by stevenireeves** [comment]
@cheshire can you re-approve to get the ball rolling again?

**Review by cheshire — APPROVED** [review]

**Comment by stevenireeves** [comment]
@gbaned can we rerun the Ubuntu-CPU test? I did not touch any CPU code in this PR, and I believe the failures to be transient.

**Comment by gbaned** [comment]
@stevenireeves  Can you please resolve conflicts? Thanks!

**Comment by stevenireeves** [comment]
@gbaned conflicts resolved.

**Comment by deven-amd** [comment]
@chsigg @cheshire gentle ping.

commits to resolve merge conflicts, are invalidating prior approvals and hence the request to approve yet again...thanks

**Comment by stevenireeves** [comment]
@cheshire can you please re-review this PR?

**Review by cheshire — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by akuegel** [comment]
The change got reverted because some internal target included cusolver.h and thus got broken by this change. But I will roll forward with the fix.

**Comment by stevenireeves** [comment]
@akuegel can you post on this PR when it is fully integrated?

**Comment by akuegel** [comment]
Will do, but probably will happen tomorrow because I am waiting for internal approval of the modified change.

**Comment by akuegel** [comment]
Your change has now been rolled forward again.

**Pull request merged**
